lib/oban/stager.ex

```diff
@@ -123,17 +123,18 @@ defmodule Oban.Stager do
   defp stage_scheduled(state, leader?: true) do
     subquery =
       Job
+      |> select([j], %{id: j.id})
       |> where([j], j.state in ["scheduled", "retryable"])
       |> where([j], not is_nil(j.queue))
+      |> where([j], j.priority >= 0 and j.priority <= 3)
```

Comment:
The `>=` and `<=` comparisons won't make use of the index like an `IN` clause:

```suggestion
      |> where([j], j.priority in [0, 1, 2, 3])
```